{
  logStore.afterPropertiesSet();
  Throwable ex=new RuntimeException("test exception");
  ex.fillInStackTrace();
  logStore.debug(repoUrl,"debug message 1");
  logStore.error(repoUrl,"error message 1",ex);
  logStore.info(repoUrl,"info message 1");
  logStore.debug(repoUrl,"debug message 2");
  logStore.error(repoUrl,"error message 2",ex);
  logStore.info(repoUrl,"info message 2");
  assertEquals(6,logStore.getLogEntries(0,10).size());
  assertEquals(5,logStore.getLogEntries(0,5).size());
  assertEquals(4,logStore.getLogEntries(2,5).size());
  assertEquals(3,logStore.getLogEntries(2,3).size());
  assertEquals(2,logStore.getLogEntries(0,10,INFO).size());
  assertEquals(1,logStore.getLogEntries(0,1,INFO,DEBUG).size());
  assertEquals(2,logStore.getLogEntries(2,4,INFO,DEBUG).size());
  assertEquals(3,logStore.getLogEntries(0,3,INFO,DEBUG).size());
}
